### PR TITLE
Library updates

### PR DESCRIPTION
**Pull request by Ghabry**
Everything updated except:
 - ~~SDL on Android (for obvious reasons)~~
 - Harfbuzz (new major version, sounds like work and patches)
 - ICU (all devkitXXX and Vita ports will need new patches)

Results:

- [x] Android
- [x] OSX
- [x] iOS
- [x] Emscripten
- [x] 3DS
- [x] Wii
- [x] Switch
- [x] Vita

### CODE REVIEW

**Comment by Ghabry** [comment]
Status:
Get strange linker errors with Harfbuzz when compiling linux-static player, but must be my environment, I didn't update freetype or harfbuzz here.

OSX fails because of linker error in SDL mixer, the patch from last time doesn't work anymore, after autoreconf -fi I get libtool version mismatch errors.

3DS and Wii can't be tested by me because the devkitNIH bootstrap script is broken.

Same for Emscripten: Deployment scripts don't work.

File: shared/packages.sh [hunk 1]
```diff
@@ -108,17 +109,18 @@ ICUDATA_URL=https://ci.easyrpg.org/job/icudata/lastSuccessfulBuild/artifact/icud
...
 lib=SDL2_mixer
-ver=2.0.1
+ver=2.0.4
 SDL2_MIXER_URL="https://www.libsdl.org/projects/SDL_mixer/release/$lib-$ver.tar.gz"
 SDL2_MIXER_DIR="$lib-$ver"
 SDL2_MIXER_ARGS="--with-sdl-prefix=$WORKSPACE --disable-music-ogg \
 	--disable-music-midi-fluidsynth --disable-music-midi-fluidsynth-shared \
-	--disable-music-mod --disable-music-mp3 --disable-music-flac --disable-sdltest"
+	--disable-music-mod --disable-music-mp3 --disable-music-flac --disable-sdltest \
+	--disable-music-opus --disable-music-mp3-mpg123"
```
Comment (Ghabry) [inline shared/packages.sh]: disable-music-mp3 doesn't disable mpg123 ¯\\\_(ツ)_/¯

File: osx/2_build_toolchain.sh [hunk 2]
```diff
@@ -62,8 +63,6 @@ export MAKEFLAGS="-j${nproc:-2}"
 
 set_build_flags
 
-install_lib $ICU_DIR/source $ICU_ARGS
-
 # Install libraries
```
Comment (Ghabry) [inline osx/2_build_toolchain.sh]: building ICU was twice in the OSX script (wups)

File: shared/packages.sh [hunk 3]
```diff
@@ -6,13 +6,13 @@ ZLIB_URL="https://zlib.net/$lib-$ver.tar.gz"
 ZLIB_DIR="$lib-$ver"
 
 lib=libpng
-ver=1.6.34
-LIBPNG_URL="https://ftp-osl.osuosl.org/pub/libpng/src/archive/xz/libpng16/$lib-$ver.tar.xz"
+ver=1.6.35
+LIBPNG_URL="https://download.sourceforge.net/libpng/$lib-$ver.tar.xz"
```
Comment (carstene1ns) [inline shared/packages.sh]: I thought we wanted to not rely on sf.net anymore.

Comment (Ghabry) [reply shared/packages.sh]: forgot to comment on this: The ftp mirror is outdated and doesn't have that file ;)

Comment (fdelapena) [reply shared/packages.sh]: libpng 1.6.36 released in the meanwhile, but provides a performance gain for ARM, maybe is worth to get a bump.

Comment (Ghabry) [reply shared/packages.sh]: still sourceforge only and they removed the dead mirror completely 👎

**Comment by carstene1ns** [comment]
also pixman 0.36 released in the meanwhile.

**Comment by Ghabry** [comment]
Are you aware of any other library updates besides pixman?

I will also try to compile devkitARM and devkitPPC again, maybe Wtrmute fixed the build by now.

**Comment by carstene1ns** [comment]
freetype 2.9.1

**Comment by Ghabry** [comment]
> ./emsdk install sdk-master-64bit

That line doesn't compile anymore on newer gcc-versions

> ./emsdk install sdk-incoming-64bit

works

**Comment by Ghabry** [comment]
This should be ready, won't touch Harfbuzz as patching this requires some work.

I submitted a patch for the SDL Wii fix upstream.

**Pull request merged**